labs/8.md

```diff
@@ -1,9 +1,103 @@
-## Conclusion
+# Convert a Namespace to a Federated Namespace
+Applications within OpenShift may already be deployed in an unfederated way. The `kubefedctl` tool will allow for a namespace and the objects within that namespace to become federated objects. This lab will deploy an application with a deployment and service and then use kubefedctl to federate the application and have it running on all three clusters. 
 
-Thanks for attending this hands on lab. By now you have completed the following activities.
+## Create a Namespace
+To start with we will create a namespace for the *reverse-words* application to be deployed.
 
-* Explored the environment
-* Deployed KubeFed
-* Flattened out your configuration files
-* Deployed 2 applications
-* Etc...
+~~~sh
+oc create ns --context cluster1 reverse-words
+~~~
+
+Now that the namespace exists deploy the deployment and service objects.
+~~~sh
+cd labs/reverse-yaml
+oc create -n reverse-words --context cluster1 -f 01-service.yaml
+oc create -n reverse-words --context cluster1 -f 02-deployment.yaml
+oc expose -n reverse-words --context cluster1 svc/reverse-words
+~~~
+
+Verify the application is running.
+~~~sh
+oc get all -n reverse-words --context cluster1
+NAME                                 READY   STATUS    RESTARTS   AGE
+pod/reverse-words-7fc48b5bd4-h9qlv   1/1     Running   0          14s
+
+NAME                    TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
+service/reverse-words   ClusterIP   172.31.2.137   <none>        8080/TCP   16s
+
+NAME                            READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/reverse-words   1/1     1            1           14s
+
+NAME                                       DESIRED   CURRENT   READY   AGE
+replicaset.apps/reverse-words-7fc48b5bd4   1         1         1       14s
+
+NAME                                     HOST/PORT                                               PATH   SERVICES        PORT   TERMINATION   WILDCARD
+route.route.openshift.io/reverse-words   reverse-words-reverse-words.apps.cluster-1.sysdeseng.com          reverse-words   8080                 None
+~~~
+
+Use the oc client to display the route for the *reverse-words* application and use curl to reverse a word.
+~~~sh
+oc get route -n reverse-words --context cluster1
+curl http://reverse-words-reverse-words.apps.cluster-1.sysdeseng.com/ -X POST -d '{"word": "PALC"}'
```

Comment:
I'd suggest changing this to:

curl http://$(oc get route -n reverse-words --context cluster1 reverse-words -o jsonpath='{.status.ingress[*].host}') -X POST -d '{"word": "PALC"}'

Reply:
I think the only hesitation that I have is removing any of the day to day stuff that an admin or user would expect to see vs scripting the whole thing. I agree with scripting there is less room for error but I just worry the user may miss things that they normally would do. Also, i worry that it just encourages them to copy and paste

Reply:
Fair enough, let's keep it that way then.